src/ripple/app/misc/TxQ.h

```diff
@@ -149,9 +158,19 @@ class TxQ
         amendment is not enabled, OR if the account has no transactions
         in the queue.
     */
-    boost::optional<std::map<TxSeq, AccountTxDetails>>
+    boost::optional<std::map<TxSeq, AccountTxDetails const>>
     getAccountTxs(AccountID const& account, ReadView const& view) const;
 
+    /** Returns information about all transactions currently
+        in the queue.
+
+        @returns Uninitialized @ref optional if the FeeEscalation
+        amendment is not enabled, OR if there are no transactions
+        in the queue.
+    */
+    boost::optional<std::vector<TxDetails>>
```

Comment:
I accidentally deleted my original reply, so I'll summarize: In addition to avoiding instantiation, the other reason was that I was following the pattern from 
```
    boost::optional<Metrics>
    getMetrics(OpenView const& view,
        std::uint32_t txCountPadding = 0) const;
```
where it made sense. It's really not necessary for `map` and `vector`, so I'll push a change soon.